### PR TITLE
OIDC logs

### PR DESCRIPTION
**Pull request by moticless**
#### What does this PR do?
Adding some basic logs to authn_oidc flow

#### Any background context you want to provide?
no

#### What ticket does this PR close?
Connected to #947

#### Where should the reviewer start?
any

#### How should this be manually tested?
Run basic authn_oidc flow

### CODE REVIEW

File: app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb [hunk 1]
```diff
@@ -47,6 +47,8 @@ def decoded_id_token
             @id_token_jwt,
             certs
           )
+          Rails.logger.debug("[OIDC] Decode ID Token succeeded")
+          @decoded_id_token
```
Comment (orenbm) [inline app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb]: this should be `decoded_id_token`

File: app/domain/authentication/authn_oidc/authenticate_id_token/provider_certificate.rb [hunk 2]
```diff
@@ -7,6 +7,8 @@ module AuthenticateIdToken
       class ProviderCertificate
 
         def fetch_certs(provider_uri)
+          Rails.logger.debug("[OIDC] Fetching provider certificate from '#{@provider_uri}'")
```
Comment (orenbm) [inline app/domain/authentication/authn_oidc/authenticate_id_token/provider_certificate.rb]: this should be `provider_uri`

File: app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb (hunk 1 above)
Comment (orenbm) [reply app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb]: fixed it by adding a method
```
def decode_id_token
          decoded_id_token
          Rails.logger.debug("[OIDC] Decode ID Token succeeded")
end
```

which will be called in `call`

File: app/domain/authentication/authn_oidc/authenticate_id_token/provider_certificate.rb (hunk 2 above)
Comment (orenbm) [reply app/domain/authentication/authn_oidc/authenticate_id_token/provider_certificate.rb]: fixed it

File: app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb [hunk 3]
```diff
@@ -12,14 +12,21 @@ module AuthenticateIdToken
       ) do
 
         def call
-          certs # may raise errors
-          decoded_id_token # we decode the id token separately to propagate relevant errors
+          # we fetch the certs & decode the id token here to propagate relevant errors
+          certs
```
[the remark this answers is not in the document]
Comment (orenbm) [reply app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb]: this means that the method will be
```
        def fetch_certs
          @fetch_certs ||= @provider_certificate.fetch_certs(@provider_uri)
        end
```

so the memoized object will also be action-oriented. wdyt?

[the remark this answers is not in the document]
Comment (jtuttle) [reply app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb]: I would leave that as `@certs`. It doesn't have to match.

[the remark this answers is not in the document]
Comment (orenbm) [reply app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb]: oh. that was the case in the beginning and code climate yelled at me. would you ignore that?

[the remark this answers is not in the document]
Comment (jtuttle) [reply app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb]: Aah I see. Well I disagree with a lot of CodeClimate's rules so yeah that's one that I would ignore personally.

**Pull request merged**